src/upload.py

```diff
@@ -0,0 +1,33 @@
+import os
+from werkzeug.utils import secure_filename
+from flask import redirect, request, flash
+
+
+ALLOWED_EXTENSIONS = {'png', 'jpg', 'jpeg', 'gif'}
+
+
+def allowed_file(filename):
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+def update_avatar(db, username):
+    if 'file' not in request.files:
+        flash('No file part')
+    file = request.files['file']
+    if file.filename == '':
+        flash('No selected file')
```

Comment:
I'll change here some flash things if you don't mind)

Reply:
Yes, thank you!